### PR TITLE
[FIX] sale_coupon: Prevent already encoded-expired coupons to be used

### PR DESCRIPTION
**Pull request by tivisse**
TaskID: 2439224

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Williambraecky — APPROVED** [review]
LGTM, this needs to be done in the refactoring aswell.

File: addons/sale_coupon/models/sale_order.py [hunk 1]
```diff
@@ -393,6 +393,11 @@ def _remove_invalid_reward_lines(self):
         # delete reward line coming from an archived coupon (it will never be updated/removed when recomputing the order)
         invalid_lines = order.order_line.filtered(lambda line: line.is_reward_line and line.product_id.id not in reward_product_ids)
 
+        # Remove coupons that are now expired
+        expired_coupons = order.applied_coupon_ids.filtered(
+            lambda coupon: coupon.expiration_date < fields.Date.today())
```
Comment (Feyensv) [inline addons/sale_coupon/models/sale_order.py:398]:
```suggestion
            lambda coupon: coupon.expiration_date and coupon.expiration_date < fields.Date.today())
```
?

Comment (Williambraecky) [reply addons/sale_coupon/models/sale_order.py:398]: Actually
```suggestion
            lambda coupon: not coupon.expiration_date or coupon.expiration_date < fields.Date.today())
```

**Comment by tivisse** [comment]
@robodoo r+

**Comment by Williambraecky** [comment]
@tivisse seems like the r+ was not taken into account.

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**